src/core/immerClass.ts

```diff
@@ -194,20 +212,30 @@ export class Immer implements ProducersFns {
 			applyPatchesImpl(draft, patches)
 		)
 	}
+
+	existingProxyMap_: WeakMap<Objectish, Drafted> | null = null
```

Comment:
Was testing the behavior in my own tool and realized that, when a draft is finalized, the two objects will be different objects rather than the same object. I'll need to add a `draftToStateMap_` as well.